common/src/main/java/com/wynntils/screens/itemfilter/ItemFilterScreen.java

```diff
@@ -0,0 +1,1279 @@
+                listener.mouseReleased(adjustedMouseX, adjustedMouseY, button);
+            }
+        }
+
+        draggingProviderScroll = false;
+        draggingSortScroll = false;
+
+        return true;
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double deltaX, double deltaY) {
+        double adjustedMouseX = mouseX - translationX;
+        double adjustedMouseY = mouseY - translationY;
+        double scrollValue = -Math.signum(deltaY);
+
+        for (GuiEventListener listener : this.children) {
+            if (listener.mouseScrolled(adjustedMouseX, adjustedMouseY, deltaX, deltaY)) {
+                return true;
+            }
+        }
+
+        if (presets.size() > MAX_PRESETS && adjustedMouseX >= Texture.ITEM_FILTER_BACKGROUND.width()) {
+            scrollPresets((int) scrollValue);
+            return true;
+        }
+
+        if (sortMode && adjustedMouseX >= 150) {
+            if (Math.abs(deltaY) == 1.0) {
+                scrollSorts((int) -deltaY);
+                return true;
+            }
+
+            // Account for scrollpad
+            currentUnusedSortScroll -= deltaY / 5d;
+
+            if (Math.abs(currentUnusedSortScroll) < 1) return true;
+
+            int scroll = (int) (currentUnusedSortScroll);
+            currentUnusedSortScroll = currentUnusedSortScroll % 1;
+
+            scrollSorts(scroll);
+        } else {
+            if (Math.abs(deltaY) == 1.0) {
+                scrollProviders((int) -deltaY);
+                return true;
+            }
+
+            // Account for scrollpad
+            currentUnusedProviderScroll -= deltaY / 5d;
+
+            if (Math.abs(currentUnusedProviderScroll) < 1) return true;
+
+            int scroll = (int) (currentUnusedProviderScroll);
+            currentUnusedProviderScroll = currentUnusedProviderScroll % 1;
+
+            scrollProviders(scroll);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean charTyped(char codePoint, int modifiers) {
+        return focusedTextInput != null && focusedTextInput.charTyped(codePoint, modifiers);
+    }
+
+    @Override
+    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+        if (keyCode == GLFW.GLFW_KEY_ESCAPE && this.shouldCloseOnEsc()) {
+            onClose();
+            return true;
+        }
+
+        return focusedTextInput != null && focusedTextInput.keyPressed(keyCode, scanCode, modifiers);
+    }
+
+    @Override
+    public TextInputBoxWidget getFocusedTextInput() {
+        return focusedTextInput;
+    }
+
+    @Override
+    public void setFocusedTextInput(TextInputBoxWidget focusedTextInput) {
+        this.focusedTextInput = focusedTextInput;
+    }
+
+    public void addFilter(Pair<ItemStatProvider<?>, String> newFilter) {
+        filters.add(newFilter);
+        usedProviders.add(newFilter.a());
+
+        createFilterQuery();
+        populateQuery();
```

Comment:
Do you not need to call `populateSorts` here and in `removeFilter`? If not, maybe you could include a comment about them below the two calls.

Reply:
No, because those 2 methods can only be called from the value widgets so you won't be viewing the sorts at the moment

Reply:
If you could add a comment in the code, we can mark this as completed.